webapp/channels/src/components/cloud_preview_modal/preview_modal_controller.test.tsx

```diff
@@ -0,0 +1,248 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {render, screen, fireEvent} from '@testing-library/react';
+import React from 'react';
+import {IntlProvider} from 'react-intl';
+
+import type {PreviewModalContentData} from './preview_modal_content_data';
+import PreviewModalController from './preview_modal_controller';
+
+// Mock the GenericModal component
+jest.mock('@mattermost/components', () => ({
```

Comment:
Why mock the modal and icon components? Those are dumb enough that I wouldn't have expected us to need to mock those. The same goes for the MM logo component in the other test